invenio_db/shared.py

```diff
@@ -29,9 +33,34 @@
 """Default database metadata object holding associated schema constructs."""
 
 
+class UTCDateTime(TypeDecorator):
+    """Custom UTC datetime type."""
+
+    impl = DateTime
+
+    def process_bind_param(self, value, dialect):
+        """Process value storing into database."""
+        if isinstance(value, datetime):
+            return value.replace(tzinfo=None)
```

Comment:
minor: I think here we should also check that `value.tzinfo in (None, timezone.utc)`, and only in that case proceed. I would even go as far as raising an exception if it's not the case, since implicitly dropping the timezone info (or converting from local to UTC) is the behavior we want to avoid.

Needs to be tested in a couple of complex modules first (e.g. `invenio-rdm-records`), but I have a feeling that it'll also help us to catch some datetime-related bugs :)